source/simulator/helper_functions.cc

```diff
@@ -286,7 +286,134 @@ namespace aspect
 
 
   template <int dim>
-  std::pair<double,bool> Simulator<dim>::compute_time_step () const
+  bool Simulator<dim>::maybe_do_initial_refinement (const unsigned int max_refinement_level)
+  {
+    if (pre_refinement_step < parameters.initial_adaptive_refinement)
+      {
+        if (parameters.timing_output_frequency ==0)
+          computing_timer.print_summary ();
+
+        output_statistics();
+
+        if (parameters.run_postprocessors_on_initial_refinement)
+          postprocess ();
+
+        refine_mesh (max_refinement_level);
+        ++pre_refinement_step;
+        return true;
+      }
+    else
+      {
+        // invalidate the value of pre_refinement_step since it will no longer be used from here on
+        pre_refinement_step = std::numeric_limits<unsigned int>::max();
+        return false;
+      }
+  }
+
+
+
+  template <int dim>
+  void Simulator<dim>::maybe_refine_mesh (const double new_time_step,
+                                          unsigned int &max_refinement_level)
+  {
+    /*
+     * see if this is an additional refinement cycle. An additional refinement
+     * cycle differs from a regular, because the maximal refinement level allowed
+     * is increased by one from this time on.
+     */
+    if ((parameters.additional_refinement_times.size() > 0)
+        &&
+        (parameters.additional_refinement_times.front () < time+new_time_step))
+      {
+        // loop over as many times as this is necessary
+        while ((parameters.additional_refinement_times.size() > 0)
+               &&
+               (parameters.additional_refinement_times.front () < time+new_time_step))
+          {
+            ++max_refinement_level;
+            refine_mesh (max_refinement_level);
+
+            parameters.additional_refinement_times
+            .erase (parameters.additional_refinement_times.begin());
+          }
+      }
+    // see if this is a time step where regular refinement is requested
+    else if (
+      (timestep_number > 0
```

Comment:
let's start the actual condition on the line with the `if` as we do in other places. if you want to offset for better readability, add a couple of spaces